apps/services/user-notification/src/app/modules/notifications/tests/notifications.controller.spec.ts

```diff
@@ -0,0 +1,143 @@
+      const templates: HnippTemplate[] = [
+        {
+          templateId: 'HNIPP.POSTHOLF.NEW_DOCUMENT',
+          notificationTitle: 'New document',
+          notificationBody: 'New document from {{organization}}',
+          args: ['arg1', 'arg2'],
+        },
+      ]
+
+      jest
+        .spyOn(notificationsService, 'getTemplates')
+        .mockResolvedValue(templates)
+
+      const locale = 'is'
+      const result = await controller.getNotificationTemplates(locale)
+
+      expect(notificationsService.getTemplates).toHaveBeenCalledWith(locale)
+      expect(result).toEqual(templates)
+    })
+  })
+
+  describe('getNotificationTemplate', () => {
+    it('should return a single notification template by ID', async () => {
+      const template: HnippTemplate = {
+        templateId: 'HNIPP.POSTHOLF.NEW_DOCUMENT',
+        notificationTitle: 'Title',
+        notificationBody: 'Body',
+        args: ['arg1'],
+      }
+
+      jest
+        .spyOn(notificationsService, 'getTemplate')
+        .mockResolvedValue(template)
+
+      const result = await controller.getNotificationTemplate(
+        'HNIPP.POSTHOLF.NEW_DOCUMENT',
+        'is',
+      )
+
+      expect(notificationsService.getTemplate).toHaveBeenCalledWith(
+        'HNIPP.POSTHOLF.NEW_DOCUMENT',
+        'is',
+      )
+      expect(result).toEqual(template)
+    })
+  })
+  describe('createHnippNotification', () => {
+    it('should validate and queue a new Hnipp notification, returning a response with the id', async () => {
+      const createHnippNotificationDto: CreateHnippNotificationDto = {
+        recipient: '1234567890',
+        templateId: 'HNIPP.POSTHOLF.NEW_DOCUMENT',
+        args: [
+          { key: 'organization', value: 'Test Organization' },
+          { key: 'documentId', value: '1234' },
+        ],
+      }
+
+      jest.spyOn(notificationsService, 'validate').mockResolvedValue(undefined)
+
+      const mockQueueId = 'mockQueueId'
+      jest.spyOn(queueService, 'add').mockResolvedValue(mockQueueId)
+
+      let validationError
+      try {
+        await controller.createHnippNotification(createHnippNotificationDto)
+      } catch (error) {
+        validationError = error
+      }
+
+      // Asserting the validation passed by checking no error was thrown
+      expect(validationError).toBeUndefined()
+
+      // Additionally, checking the validate method was called correctly
+      expect(notificationsService.validate).toHaveBeenCalledWith(
+        createHnippNotificationDto.templateId,
+        createHnippNotificationDto.args,
+      )
+
+      // Assert that the queueService.add was called with any argument, considering
+      // the test setup does not specify the exact argument structure.
+      expect(queueService.add).toHaveBeenCalledWith(expect.anything())
+
+      // If reaching this point without errors, validation is considered successful.
+      expect(true).toBe(true) // This is implicitly true if no error was thrown.
```

Comment:
Is it possible to utilize `expect().not.toThrow(error)` on the try/catch block?
And, isn't this already being done, sort of, with `expect(validationError).toBeUndefined()`?